### PR TITLE
902 trap focus in drawer

### LINKED ISSUE #902: o-header: user can leave open drawer accidentally and navigate content behind
When a user of screen reading assistive technologies navigates through the expanded menu using the arrow keys, they are not trapped within the menu and can leave the menu returning to the page. However, this only occurs when navigating back up the page, as the menu has been situated at the end of the DOM. 

*See how we trap focus in o-overlay*

https://docs.google.com/document/d/17-d7DAAORFCaUbiHdFPqvrBerUKbMyhG/edit#bookmark=id.8nyxjz95mzek
https://financialtimes.atlassian.net/browse/CON-2116

image-20221012-105433

### PR DESCRIPTION
**Pull request by akomiqaia**
closes #902

### CODE REVIEW

**Comment by notlee** [comment]
@akomiqaia This solves the problem but makes a new one! 
1. `display: none` is not set by default, until the drawer is opened and closed. We could use `.o-header__drawer[data-o-header-drawer--js][aria-hidden]` instead to apply only when the JavaScript is initialised and the drawer is hidden?
2. Setting `display: none` breaks the drawer's animation. We set `aria-hidden`, is that enough? Do we really need to set `display: none`? If we do, perhaps we can [toggle display based on transition events](https://dev.to/tylerlwsmith/leveraging-javascript-to-implement-css-transitions-that-use-display-none-4hhb)?

**Comment by akomiqaia** [comment]
@notlee I tried to get this transition working. It took me some time to make it work and it is still pretty strange how it works. I tired to apply `display: none` on the drawer element itself but that stops `transitionend` from firing. So instead children are set `display:none' and just before we toggle the drawer to open we remove the display attribute to allow transitions happening smoothly.

File: components/o-header/src/js/drawer.js [hunk 1]
```diff
@@ -82,6 +89,7 @@ function handleCloseEvents (scope, callback, allFocusable) {
 		document.addEventListener('focusout', handleFocus);
 
 		scope.addEventListener('keydown', handleTab);
+		scope.addEventListener('transitionend', handleTransitionEnd);
```
Comment (notlee) [inline components/o-header/src/js/drawer.js:92]: This should be removed in `removeEvents` above else will be added multiple times.

File: components/o-header/demos/src/sticky-header.mustache [hunk 2]
```diff
@@ -87,7 +87,7 @@
 	</header>
 
 	{{#drawer}}
-	<div class="o-header__drawer" id="o-header-drawer" role="navigation" aria-label="Drawer menu" data-o-header-drawer data-o-header-drawer--no-js>
+	<div class="o-header__drawer" id="o-header-drawer" aria-modal=true role="dialog" aria-label="Drawer menu" data-o-header-drawer data-o-header-drawer--no-js>
```
Comment (notlee) [inline components/o-header/demos/src/sticky-header.mustache:90]: Works great :) We also need to update the migration guide to instruct users to make these markup changes to the drawer.

Comment (akomiqaia) [reply components/o-header/demos/src/sticky-header.mustache:90]: @notlee [73d8a96](https://github.com/Financial-Times/origami/pull/939/commits/73d8a96445b69bec5f2d379197b0334228418955) add more details on migration

File: components/o-header/MIGRATION.md [hunk 3]
```diff
@@ -46,6 +46,10 @@ Remove the CSS class `o-header__drawer-menu-item--divide` from the last list ite
 
 For consistency and accessibility label for drawer's close button is `Close side navigation menu`. Changes can be observed in this [PR](https://github.com/Financial-Times/origami/pull/903). You might need to update your markup accordingly.
 
+## Update ARIA attributes
+
+`.o-header__drawer` now has `aria-modal="true"` and `role="navigation"` attribute was updated to `role="modal"`. [This change](https://github.com/Financial-Times/origami/pull/939) improves the experience for users of assistive technologies. You might need to update your markup accordingly.
```
[the remark this answers is not in the document]
Comment (akomiqaia) [reply components/o-header/MIGRATION.md]: [92aeecd](https://github.com/Financial-Times/origami/pull/939/commits/92aeecd8f156f963de41c792339457269cf918d7) chatGPT helped with this one :D

**Review by notlee — APPROVED** [review]

File: components/o-header/MIGRATION.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (notlee) [reply components/o-header/MIGRATION.md]: haha, with the previous or updated version!?

[the remark this answers is not in the document]
Comment (notlee) [reply components/o-header/MIGRATION.md]: chatGPT blows my mind!

**Pull request merged**